### DISCUSSION TITLE
Rename Zip::apply to Zip::for_each? [Solved]

### DISCUSSION THREAD

**Discussion by bluss** [rust-ndarray/ndarray#858] [Ideas]
Rename Zip::apply to Zip::for_each (and par_apply to par_for_each).

This seems like a suggestion that can cause some churn, (unnecessarily?), but I think the code is easier to read and write if we use the more established name `.for_each()`.

For example, new code after this change:

```rust
Zip::from(result.index_axis_mut(axis, perm_i))
    .and(self.index_axis(axis, i))
    .for_each(|to, from| {
        copy_nonoverlapping(from, to.as_mut_ptr(), 1);
        moved_elements += 1;
    });
```

The old name would be deprecated and removed a release later. At the moment, there is no way for us to integrate with "cargo fix" (only rustc_deprecated attribute can do that), so we can't have this change be applied automatically.

**Comment by bluss** [comment]
@jturner314 This touches a bit too many methods to be fun (apply and par_apply of all arities) but also seems like a no-brainer. The unfortunately named `apply_collect` would more reasonably be `map_collect` I think, too.

**Comment by jturner314** [comment]
I agree that this is something we should do. Also, `apply_assign_into` should be renamed to `map_assign_into`. It's easy for users to update their code with simple renames like this (especially if the old names are deprecated for a while), so IMO consistency with `std` far outweighs the cost of updating code.
